File: Python codes/Getting_possible_solution.py
arr = [0]*16
arr[0]=1
a = [3,5,10]
for i in range(3):
    for j in range(a[i],16):
        if(arr[j-a[i]]!=0):
            arr[j] = arr[j]+arr[j-a[i]]
    
print(arr[15])

File: C codes/Getting_possible_solution.c
#include<stdio.h>
#include<string.h>

int main()
{
    int arr[16];
    int i,j,k;
    int a[]={3,5,10};
    memset(arr,0,16*sizeof(int));
    arr[0]=1;
    for(j=0;j<3;j++)
    {
        k = a[j];
        for(i=k;i<=15;i++)
        {
            if(arr[i-k]!=0)
            {
                arr[i] = arr[i] + arr[i-k];
            }
        }
    }
    printf("%d\n",arr[15]);
    return 0;
}